sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala

```diff
@@ -123,4 +126,95 @@ class SparkSessionBuilderSuite extends SparkFunSuite {
       session.stop()
     }
   }
+
+  test("fork new session and inherit a copy of the session state") {
```

Comment:
this test doesnt test anything other than `ne` which is already covered by other tests